packages/ui/src/components/ProjectCard/ProjectCard.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { Box, Flex, forwardRef, Grid } from '@chakra-ui/react';
+
+import { Button } from '../Button/Button';
+import { Typography } from '../Typography/Typography';
+
+export type StatusProps = 'idle' | 'doing' | 'review' | 'done';
+
+type RemoveCommonValues<Type, TOmit> = {
+  [Property in keyof Type]: TOmit extends Record<Property, infer U> ? Exclude<Type[Property], U> : Type[Property];
+};
+
+type Id<Type> = Record<string, unknown> & { [P in keyof Type]: Type[P] };
+
+type ConditionalProps<Type, TKey extends keyof TCase, TCase extends Partial<Type>> =
+  | Id<Omit<Type, keyof TCase> & TCase>
+  | Id<RemoveCommonValues<Type, Pick<TCase, TKey>>>;
+
+export type ProjectCardType = ConditionalProps<
+  {
+    status: StatusProps;
+    image: string;
+    title: string;
+    url: string;
+  },
+  'status',
+  | {
+      status: 'idle' | 'doing' | 'review';
+      date: Date;
+    }
+  | {
+      status: 'done';
+      points: number;
+      pointsMax: number;
+    }
+>;
```

Comment:
🤔 I think this could be achieved easier if you create a type like below
```ts
type DoneProps = {
    status: 'done';
    points: number;
    pointsMax: number;
}

type NotDoneProps = {
    status: 'idle' | 'doing' | 'review';
    date: Date;
}

export type ProjectCardProps = {
    image: string;
    title: string;
    url: string;
}  & (DoneProps | NotDoneProps)
```

Reply:
I think you have right. I'll change it.